x-pack/plugin/apm-data/src/main/resources/ingest-pipelines/traces-apm@pipeline.yaml

```diff
@@ -30,10 +30,10 @@ processors:
     field: ["event.duration"]
     ignore_failure: true
     ignore_missing: true
-- set:
-    if: ctx.event?.outcome == 'success'
-    field: event.success_count
-    value: 1
+- script:
+    if: ctx.event?.outcome == 'success' && ctx[ctx.processor?.event]?.representative_count != null
+    source: |
+        ctx.event.success_count = ctx[ctx.processor?.event]?.representative_count;
```

Comment:
@rubvs the implementation is different than what was suggested by @lahsivjar in the github card:
>The issue is about fixing the ingest pipeline to use representative count if defined, 1 otherwise in the ingest pipeline for cases when the event.outcome is success.

If `representative_count` is not available, `event.count` should still be populated according to `event_outcome`.